ompi/mca/coll/tuned/coll_tuned_decision_fixed.c

```diff
@@ -217,6 +217,122 @@ ompi_coll_tuned_allreduce_intra_dec_fixed(const void *sbuf, void *rbuf, int coun
                                                     comm, module, alg, 0, 0);
 }
 
+
+/*
+ *  allreduce_intra for single node communicators
+ *
+ *  Function:   - allreduce using other MPI collectives
+ *  Accepts:    - same as MPI_Allreduce()
+ *  Returns:    - MPI_SUCCESS or error code
+ */
+int
+ompi_coll_tuned_allreduce_intra_singlenode_dec_fixed(const void *sbuf, void *rbuf, int count,
+                                                     struct ompi_datatype_t *dtype,
+                                                     struct ompi_op_t *op,
+                                                     struct ompi_communicator_t *comm,
+                                                     mca_coll_base_module_t *module)
+{
+    size_t dsize, total_dsize;
+    int communicator_size, alg;
+    communicator_size = ompi_comm_size(comm);
+    OPAL_OUTPUT((ompi_coll_tuned_stream, "ompi_coll_tuned_allreduce_intra_singlenode_dec_fixed"));
+
+    ompi_datatype_type_size(dtype, &dsize);
+    total_dsize = dsize * (ptrdiff_t)count;
+
+    /** Algorithms:
+     *  {1, "basic_linear"},
+     *  {2, "nonoverlapping"},
+     *  {3, "recursive_doubling"},
+     *  {4, "ring"},
+     *  {5, "segmented_ring"},
+     *  {6, "rabenseifner"
+     *
+     * Currently, ring, segmented ring, and rabenseifner do not support
+     * non-commutative operations.
+     */
+    if( !ompi_op_is_commute(op) ) {
+        if (communicator_size < 4) {
+            alg = 1;
+        } else if (communicator_size < 8) {
+            if (total_dsize < 128) {
+                alg = 1;
+            } else if (total_dsize < 16384) {
+                alg = 3;
+            } else if (total_dsize < 262144) {
+                alg = 2;
+            } else {
+                alg = 1;
+            }
+        } else if (communicator_size < 16) {
+            if (total_dsize < 128) {
+                alg = 1;
+            } else if (total_dsize < 4096) {
+                alg = 3;
+            } else if (total_dsize < 262144) {
+                alg = 2;
+            } else {
+                alg = 1;
+            }
+        } else {
+            if (total_dsize < 4096) {
+                alg = 3;
+            } else {
+                alg = 2;
+            }
+        }
+        /* TODO: evaluate allreduce with non-commutative ops */
+        return ompi_coll_tuned_allreduce_intra_dec_fixed(sbuf, rbuf, count,
```

Comment:
Why are you setting alg if you go with decision fixed ?

Reply:
Oops, that was a leftover from before I had non-commutative ops. Fixed